### PR TITLE
Jump search

### PR DESCRIPTION
**Pull request by BriseBalloches**
### **Describe your change:**

Added type hinting, docstring, doctest for jump search algorithm.

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [X] Documentation change?

### **Checklist:**
* [X] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [X] This pull request is all my own work -- I have not plagiarized.
* [X] I know that pull requests will not be merged if they fail the automated tests.
* [X] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [X] All filenames are in all lowercase characters with no spaces or dashes.
* [X] All functions and variable names follow Python naming conventions.
* [X] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [X] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: searches/jump_search.py [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+until the element compared is bigger than the one searched.
+It will then perform a linear search until it matches the wanted number.
+If not found, it returns -1.
+"""
+
 import math
 
 
-def jump_search(arr, x):
+def jump_search(arr: list, x: int) -> int:
```
Comment (ruppysuppy) [inline searches/jump_search.py:12]:
```suggestion
def jump_search(arr: List[int], x: int) -> int:
```
First import List from typing

**Review by ruppysuppy — CHANGES_REQUESTED** [review]

File: searches/jump_search.py [hunk 2]
```diff
@@ -21,6 +42,7 @@ def jump_search(arr, x):
 
 
 if __name__ == "__main__":
-    arr = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610]
-    x = 55
+    user_input = input("Enter numbers separated by a comma:\n").strip()
+    arr = [int(item) for item in user_input.split(",")]
+    x = int(input("Enter the number to be searched:\n"))
     print(f"Number {x} is at index {jump_search(arr, x)}")
```
Comment (ruppysuppy) [inline searches/jump_search.py:48]: -1 is a valid position in python, check if the number is found before printing it

**Review by ruppysuppy — CHANGES_REQUESTED** [review]

**Comment by BriseBalloches** [comment]
Modified. Thanks.

File: searches/jump_search.py (hunk 1 above)
Comment (cclauss) [reply searches/jump_search.py:12]: This limits us to only the __int__ data type.  Without this specificity, would the algorithm work with strings, tulles, floats?

File: searches/jump_search.py [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
+def jump_search(arr: list, x: int) -> int:
+    """
+    Pure Python implementation of the jump search algorithm.
+    Examples:
+    >>> jump_search([0, 1, 2, 3, 4, 5], 3)
+    3
+    >>> jump_search([-5, -2, -1], -1)
+    2
+    >>> jump_search([0, 5, 10, 20], 8)
+    -1
```
Comment (cclauss) [inline searches/jump_search.py:21]: I tend to like to raise an exception like str.index() vs str.find() but that is a personal choice.

**Review by cclauss — APPROVED** [review]
LGTM

**Review by ruppysuppy — APPROVED** [review]

**Pull request merged**